### DISCUSSION TITLE
Item Prices

### DISCUSSION THREAD

**Discussion by cstby** [cstby/foundryvtt-world-currency-5e#26] [Ideas]
### Decision

Should we support changes to item prices? If so, what is is our technical approach?

### Context

The way DND5e handles item prices has changed. When we released v2.0.2, the DND5e system denominated all prices in RAW gold. All items had a single `price` key, whose value represented the price in RAW gold pieces. For example, a blanket (5 sp RAW) would have a price of `0.5`. 

Now, item prices have two fields: a denomination and a price. For example, a blanket (5 sp RAW) is now {value: 5 denomination: "sp"}. This means our code to support "standard currency" does not work as expected.

#### Use Case 1: Silver Standard
A user setting the silver standard will accept prices in CP, SP, and GP only. The user takes the following configuration steps:
- Rename platinum to gold
- Rename gold to silver
- Rename electrum to copper
- Hide Silver
- Hide Copper

In this case, items denominated in RAW gold are shown as denominated in silver with the correct prices. For example, a chain shirt will appear to be 50 sp. However, items denominated in RAW silver will not be shown correctly. For example, a blanket (RAW 5 sp) will appear as 5 sp, when the user would expect it to be 50 cp.

#### Use Case 2: One Fiat Currency
A user with one fiat currency will want all prices displayed in that currency. The user take the following configuration steps:
- Hide platinum, gold, electrum, and silver
- Change the name and abbreviation to "Woolong" and "￦"
- Sets Woolongs to be 1000 RAW gp.

In this case, no items will appear as the user expects. Even items denominated in copper will appear 10x more expensive than the user desires.

#### Approach 1: Cease support of item prices

How important is it to show item prices in custom currencies? For many of our users, this is a fairly important feature. If/when 5e develops a standard shopping extension (and it looks like Item Piles is the front-runner), more users  will ask for us to support shopping with custom currencies.

#### Approach 2: Render item prices without changing the underlying price

This approach is similar to the v2.0.2 approach of changing how item prices appear without changing the item objects themselves. On rendering the item sheet, we would convert the `value` from the RAW denomination to the user's selection of standard currency.

#### Approach 3: Patch the item objects

In this approach, we would change the items themselves. The advantage of this approach is that once the item's base data is correct, all downstream sheets and modules will show the item price as the user expects. However, when/how to patch the item is an area where technical discovery is needed.

**Comment by cstby** [comment]
@diarioz and @UmbraZeal, you input as users is valuable here Could you let us know whether item prices and shopping are important to you?

@kmuld I'd love to get your thoughts on whether approach 3 is feasible.

**Reply by kmuld** [reply]
Regarding approach 3, I haven't taken a look at if this is technically doable.
If it is, it would require data migrations in the event that the dnd5e package changes its currency data structure again (as seen in a recent update.)

I think approach 2 would be best.
I'll likely take a look at how to accomplish this approach.

**Reply by cstby** [reply]
Perfect. I'll take a look at Item Piles and develop a POV on what our users would ideally want in their shopping experience.

**Comment by diarioz** [comment]
Hey, I’m in fact using Item Piles to manage shopping. This feature alongside the possibility to change currencies is quite important to me.
